### ISSUE TITLE
Add weight decay option

### ISSUE THREAD

**Issue by MadLittleMods** [MadLittleMods/zig-neural-networks#19]
Add weight decay option to help reduce overfitting

### Dev notes

 - https://machinelearningmastery.com/how-to-reduce-overfitting-in-deep-learning-with-weight-regularization/
 - https://github.com/SebLague/Neural-Network-Experiments/blob/2e28014ede7ae696cdccba2783a40352200ac14b/Assets/Scripts/Training/NetworkTrainer.cs#L106
